includes/admin/class-wc-rest-payments-settings-controller.php

```diff
@@ -378,6 +384,21 @@ private function update_account_statement_descriptor( WP_REST_Request $request )
 		$this->wcpay_gateway->update_option( 'account_statement_descriptor', $account_statement_descriptor );
 	}
 
+	/**
+	 * Updates WooCommerce Payments account business name.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 */
+	private function update_account_business_name( WP_REST_Request $request ) {
```

Comment:
I am not sure I am following they "bypass" part, but I see the confusion here:

If the idea is to allow merchants to change their business name on the receipt, then the setting should be called something like `receipt_business_name`. By default it would take a business name from an account, but allow to override it. Thus, there will be an _account business name_ which is shared with Stripe account, and _receipt business name_ to put on receipts.

In this case `get_option` should be updated to properly handle this and read _receipt business name_ from settings first and fallback to an _account business name_

Reply:
> I am not sure I am following they "bypass" part

This is perhaps better understood with the example of statement_descriptor.

We have the following method that saves statement_descriptor's value in WP options:

https://github.com/Automattic/woocommerce-payments/blob/06b7f96315b2853fd043f6892c22bdaa8a4be2d1/includes/admin/class-wc-rest-payments-settings-controller.php#L399-L407

Clicking the 'Save changes' button on Settings page triggers the above method.

On line 1509 below, we are returning the value of statement_descriptor directly from Stripe account data rather than from WP options.

https://github.com/Automattic/woocommerce-payments/blob/ddbf9e4323bf85da650bbee9e49edef97fce8d98/includes/class-wc-payment-gateway-wcpay.php#L1504-L1513

So, in effect, we are saving statement_descriptor in WP options but never using the saved/cached value anywhere in code.

Reply:
Thanks for explainng, @anu-rock. That's exactly my concern: there is a code path that doesn't do what it promises. In my humble opinion, `update_account_statement_descriptor` should update `account_statement_descriptor` instead of storing value in WP options. Might be even better to implement it in `update_option` method to make it aligned with `get_option`.

Reply:
> Thanks for explainng, @anu-rock. That's exactly my concern: there is a code path that doesn't do what it promises. In my humble opinion, `update_account_statement_descriptor` should update `account_statement_descriptor` instead of storing value in WP options. Might be even better to implement it in `update_option` method to make it aligned with `get_option`.

Right. I'm in agreement with your suggestion.

CCing @vbelolapotkov who added statement_descriptor some time ago to see if he anticipated the use of statement_descriptor stored in WP options.

Reply:
It's been quite long time ago before settings rework. The save actually did update of descriptor on the server end, however, I don't remember if it was triggered here or in some other hook in update process. Have you tested if save on new settings screen changes descriptor value in Stripe dashboard?

Reply:
> Have you tested if save on new settings screen changes descriptor value in Stripe dashboard?

Yes. The value for the descriptor is saved in Stripe as well as WP options. It's the latter I'm confused about. Do we need to save it there?

> Might be even better to implement it in update_option method to make it aligned with get_option.

`update_statement_descriptor` is currently called inside `sanitize_plugin_settings`. Does it make sense to call it inside `update_option` like this?

```php
public function update_option( $key, $value = '' ) {
	if ( 'enabled' === $key ) {
		$key = static::METHOD_ENABLED_KEY;
	}
	elseif ( 'account_statement_descriptor' === $key ) {
		return $this->update_statement_descriptor( $value );
	}
	return parent::update_option( $key, $value );
}
```

Reply:
I've made another check in the code and now I think I remember how it works :) Plugin settings stored as a single option `woocommerce_woocommerce_payments_settings` in DB. If you dive deeper in how update_option method works in WC you'll see that it takes whole settings object, modifies key and calls the hook `'woocommerce_settings_api_sanitized_fields_' . $this->id` to sanitise settings option before saving updated value to DB. There is [a method](https://github.com/Automattic/woocommerce-payments/blob/1caefac6d1a5cf58f5ff455f89cc4ef21bdecddc/release/woocommerce-payments/includes/class-wc-payment-gateway-wcpay.php#L1456-L1456) does "all the magic" to make sure instead of storing value in plugin_settings option it's updated on the server and removed from settings object.
That's why common method `update_option` is used every time when we need to update statement descriptor, but it doesn't save anything to DB.